### PR TITLE
feat(j-s): Enable public prosecutors to properly handle fines

### PR DESCRIPTION
**Pull request by oddsson**
# Enable public prosecutors to properly handle fines

[Asana](https://app.asana.com/0/322488613152836/1208638260498658/f)

## What

Enable public prosecutors to properly handle fines. In cases where there is a fine the following applies 

- There is no service done in these cases and no áfrýjun
- The defendant can appeal the fine, but only has three days to do so from the ruling date, instead of 28.

This PR also gives prison admin users access to cases where there is a fine.

## Why

So that the public prosecutor can properly handle all S-cases

## Screenshots / Gifs

Screenshot 2024-11-26 at 12 53 57

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `BlueBoxWithDate` component for better date display in the overview.
	- Added support for handling fine-related appeal deadlines across various components.
	- Enhanced message definitions to support dynamic rendering based on ruling decisions.
	- Added a new message for "Kært" in reviewed cases.
	- Updated the `reviewerSubtitle` to dynamically reflect appeal status and deadlines.

- **Bug Fixes**
	- Improved access control for prison admin users based on ruling decisions.

- **Documentation**
	- Updated message IDs and default messages for clarity and consistency.

- **Style**
	- Adjusted CSS grid layout for improved responsiveness in the Review Decision component.

- **Tests**
	- Added new test cases to validate the handling of fine ruling decisions and appeal deadlines.

### CODE REVIEW

File: apps/judicial-system/api/src/app/modules/case/interceptors/case.transformer.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ import {
 import { Defendant } from '../../defendant'
 import { EventLog } from '../../event-log'
 import { Case } from '../models/case.model'
+import { FINE_APPEAL_WINDOW_DAYS } from 'libs/judicial-system/types/src/lib/indictmentCase'
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/api/src/app/modules/case/interceptors/case.transformer.ts:16]: Thanks! 🙌

File: apps/judicial-system/backend/src/app/modules/case/filters/cases.filter.ts [hunk 2]
```diff
@@ -211,7 +211,12 @@ const getPrisonAdminUserCasesQueryFilter = (): WhereOptions => {
       {
         type: indictmentCases,
         state: CaseState.COMPLETED,
-        indictment_ruling_decision: CaseIndictmentRulingDecision.RULING,
+        indictment_ruling_decision: {
+          [Op.or]: [
+            CaseIndictmentRulingDecision.RULING,
+            CaseIndictmentRulingDecision.FINE,
+          ],
+        },
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/backend/src/app/modules/case/filters/cases.filter.ts:219]: Thanks, fixed

File: apps/judicial-system/web/src/components/index.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 export { CourtCaseInfo, ProsecutorCaseInfo } from './CaseInfo/CaseInfo'
 export { default as AccordionListItem } from './AccordionListItem/AccordionListItem'
 export { default as BlueBox } from './BlueBox/BlueBox'
+export { default as BlueBoxWithDate } from './BlueBoxWithIcon/BlueBoxWithDate'
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/web/src/components/index.ts:4]: How would you test this component?

File: apps/judicial-system/web/src/components/BlueBoxWithIcon/BlueBoxWithDate.tsx [hunk 4]
```diff
@@ -133,34 +137,45 @@ const BlueBoxWithDate: FC<Props> = (props) => {
   const textItems = useMemo(() => {
     const texts = []
 
-    if (serviceRequired) {
+    if (isFine) {
       texts.push(
-        formatMessage(strings.defendantVerdictViewedDate, {
-          date: formatDate(dates.verdictViewDate ?? defendant.verdictViewDate),
+        formatMessage(strings.fineAppealDeadline, {
+          appealDeadlineIsInThePast: !defendant.isVerdictAppealDeadlineExpired,
```
Comment (gudjong) [inline apps/judicial-system/web/src/components/BlueBoxWithIcon/BlueBoxWithDate.tsx:143]: Please update memo dependancies.

**Review by gudjong — APPROVED** [review]
Minor comments

**Pull request merged**